core/src/main/java/hudson/model/RunParameterDefinition.java

```diff
@@ -164,20 +168,29 @@ public ParameterValue getDefaultParameterValue() {
         }
 
         Run<?,?> lastBuild = null;
+        Job project = getProject();
+
+        if (project == null) {
+            if (LOGGER.isLoggable(WARNING)) {
+                LOGGER.log(WARNING, "project should not be null when retrieving default parameter value",
```

Comment:
Not sure this is a suitable `WARNING`, and also pretty vague -- doesn't say which project has the parameter, and what the string parameter value that has no matching project is.

Reply:
Yeah, I am not sure what this log message is doing here. If `getProject() == null`, just return null and be done with it.